shared/actions/kbfs.desktop.js

```diff
@@ -270,11 +270,11 @@ function* openWithCurrentMountDir(openPath: string): Saga.SagaGenerator<any, any
 
   // turns '/keybase/private/alice' to 'private/alice'
   const subPath = goodPath
-    .split(path.sep)
+    .split('/')
     .slice(2)
     .join(path.sep)
 
-  let kbfsPath = yield Saga.select(state => state.config.kbfsPath)
+  let { config: { kbfsPath } }= yield Saga.select()
```

Comment:
actually i think you literally should keep state as a const and explicitly mark it as TypedState
so
```
const state: TypedState = yield Saga.select()
let { config: { kbfsPath } } = state // if you want to use destructure 
// or
let kbfsPath = state.config.kbfsPath
```

Reply:
Oh I see. Would `let { config: { kbfsPath } } : TypedState = yield Saga.select()` work and if not why?

Reply:
mmmm maybe. kinda getting noisy though. Typically your saga will need access to the store so it'll do this then it'll have some conditional logic to pull subsets of the store so just doing a const state: TypedState is a very normal pattern in our sagas. This mostly goes away with the pure helpers as thats also injected in as state (which also matches this, so if you transition over to a pure function you just delete the const state line. ). 

I think its fine to leave as you posted above

Reply:
I see. Thanks for the detailed answer!

> This mostly goes away with the pure helpers as thats also injected in as state (which also matches this, so if you transition over to a pure function you just delete the const state line. ).

Sorry for massive follow-up questions, but may I ask for an example for this? I only found one `Saga.select()` from the chat refactor link you sent me, and it's inside a generator which I guess is different.

Reply:
np, i def want to answer all your questions (but chat might be better). 
This code here is in a generator. In general we're moving away from plain generators and more to using the SafeTakeEveryPure which just injects state into a plain function (hence you not seeing select in the chat refactor.) If you search the repo though you'll see lots of examples of this

Reply:
Awesome thanks!